lib/parser/index.js

```diff
@@ -950,30 +894,36 @@ function aaInnerLoop(p, furthestBlock, formattingElement) {
         lastElement = furthestBlock,
         nextElement = p.openElements.getCommonAncestor(furthestBlock);
 
-    for (var i = 0; i < AA_INNER_LOOP_ITER; i++) {
+    for (var i = 0; ; i++) {
```

Comment:
This might look not very clean IMO because logic of stop condition is hidden inside of the body. Perhaps better to put condition here explicitly? If it's not dependant on `i`, then at least `nextElement !== formattingElement` would do.